staging/src/k8s.io/kubectl/pkg/generate/versioned/run.go

```diff
@@ -327,6 +353,22 @@ func (BasicPod) Generate(genericParams map[string]interface{}) (runtime.Object,
 	return &pod, nil
 }
 
+// parseAnnotations converts []string into map of annotations.
```

Comment:
Is there a way you can use [this](https://github.com/kubernetes/kubernetes/blob/b497fa1d34ec15c4219d591e541c2cf811271357/staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go#L380-L453) all here?